habitat/tasks/rearrange/rearrange_task.py

```diff
@@ -45,6 +49,29 @@ class RearrangeTask(NavigationTask):
     def overwrite_sim_config(self, sim_config, episode):
         return merge_sim_episode_with_object_config(sim_config, episode)
 
+    def _duplicate_sensor_suite(self, sensor_suite: SensorSuite) -> None:
+        """
+        Modifies the sensor suite in place to duplicate robot specific sensors
+        between the two robots.
+        """
+
+        task_new_sensors: Dict[str, Sensor] = {}
+        task_obs_spaces = OrderedDict()
+        for robot_idx, agent_id in enumerate(self._sim.robots_mgr.agent_names):
+            for sensor_name, sensor in sensor_suite.sensors.items():
+                if isinstance(sensor, UsesRobotInterface):
+                    new_sensor = copy.copy(sensor)
+                    new_sensor.robot_id = robot_idx
+                    full_name = f"{agent_id}_{sensor_name}"
+                    task_new_sensors[full_name] = new_sensor
+                    task_obs_spaces[full_name] = new_sensor.observation_space
+                else:
+                    task_new_sensors[sensor_name] = sensor
+                    task_obs_spaces[sensor_name] = sensor.observation_space
+
+        sensor_suite.sensors = task_new_sensors
```

Comment:
@ASzot, it looks like the copied sensor suite will *replace* all other sensor, while what we wanted to append it. For example, if our Agent_1 has other than `UsesRobotInterface` it will be ignored, while it should be appended.

Reply:
No, it is not being ignored. It is added on line 69-70 with `sensor_name`.